web/crux-ui/src/components/deployments/add-deployment-card.tsx

```diff
@@ -0,0 +1,244 @@
+import { DyoCard } from '@app/elements/dyo-card'
+import DyoChips from '@app/elements/dyo-chips'
+import { DyoHeading } from '@app/elements/dyo-heading'
+import { DyoInput } from '@app/elements/dyo-input'
+import { DyoLabel } from '@app/elements/dyo-label'
+import DyoMessage from '@app/elements/dyo-message'
+import DyoTextArea from '@app/elements/dyo-text-area'
+import { apiErrorHandler, defaultTranslator } from '@app/errors'
+import useDyoFormik from '@app/hooks/use-dyo-formik'
+import useTeamRoutes from '@app/hooks/use-team-routes'
+import {
+  CreateDeployment,
+  Deployment,
+  DyoApiError,
+  DyoNode,
+  Project,
+  Version,
+  projectNameToDeploymentPrefix,
+} from '@app/models'
+import { fetcher, sendForm } from '@app/utils'
+import { createFullDeploymentSchema } from '@app/validations'
+import useTranslation from 'next-translate/useTranslation'
+import { useEffect } from 'react'
+import useSWR from 'swr'
+
+interface AddDeploymentCardProps {
+  className?: string
+  onAdd: (deploymentId: string) => void
+  onDiscard: VoidFunction
+}
+
+const AddDeploymentCard = (props: AddDeploymentCardProps) => {
+  const { className, onAdd, onDiscard } = props
+
+  const { t } = useTranslation('deployments')
+  const routes = useTeamRoutes()
+
+  const { data: nodes, error: fetchNodesError } = useSWR<DyoNode[]>(routes.node.api.list(), fetcher)
+  const { data: projects, error: fetchProjectsError } = useSWR<Project[]>(routes.project.api.list(), fetcher)
+
+  const handleApiError = apiErrorHandler((stringId: string, status: number, dto: DyoApiError) => {
+    if (dto.error === 'rollingVersionDeployment' || dto.error === 'alreadyHavePreparing') {
+      onAdd(dto.value)
+
+      return {
+        toast: dto.description,
+        toastOptions: {
+          className: '!bg-warning-orange',
+          duration: 5000,
+        },
+      }
+    }
+
+    return defaultTranslator(t)(stringId, status, dto)
+  })
+
+  const formik = useDyoFormik({
+    initialValues: {
+      nodeId: null as string,
+      note: '',
+      prefix: '',
+      versionId: null as string,
+      projectId: null as string,
+    },
+    validationSchema: createFullDeploymentSchema,
+    onSubmit: async (values, { setSubmitting, setFieldError }) => {
+      setSubmitting(true)
+
+      const body: CreateDeployment = {
+        ...values,
+      }
+
+      const res = await sendForm('POST', routes.deployment.api.list(), body)
+
+      if (res.ok) {
+        const result = (await res.json()) as Deployment
+        onAdd(result.id)
+      } else if (res.status === 409) {
+        // Handle preparing deployment exists or rolling version has deployment errors
+        handleApiError(res.clone())
+      } else {
+        handleApiError(res, setFieldError)
+      }
+
+      setSubmitting(false)
+    },
+  })
+
+  const currentProject = projects?.find(it => it.id === formik.values.projectId)
+
+  const { data: versions, error: fetchVersionsError } = useSWR<Version[]>(
```

Comment:
Can we move this up to the other `useSWR` calls?

Reply:
No because it uses the projectId from formik to fetch the versions.